apps/aecore/lib/aecore/wallet/worker.ex

```diff
@@ -0,0 +1,84 @@
+defmodule Aecore.Wallet.Worker do
+  @defmodule """
+  Module for handling the creation of a Wallet file
+  """
+
+  use GenServer
+
+  @typedoc "Options for network"
+  @type opts :: :mainnet | :testnet
+
+  @aewallet_dir Application.get_env(:aecore, :aewallet)[:path]
+  @aewallet_pass "1234"
+
+  ## Client API
+
+  def start_link(_args) do
+    GenServer.start_link(__MODULE__, %{path: @aewallet_dir}, name: __MODULE__)
```

Comment:
Why do we need a GenServer when we don't need to keep state?

Reply:
The GenServer was used so that we can start it from the Supervisor because we need to be sure that a wallet is created before starting the miner.

Reply:
You can call the function that does this in the aecore.ex module (the start function) and get rid of the GenServer.

Reply:
Well now when I think about it, the state could store the public key of the wallet, so that the module won't need to derive it everytime it is needed.

Reply:
@gspasov I think so as well